addons/product_images/wizard/product_fetch_image_wizard.py

```diff
@@ -0,0 +1,264 @@
+        nb_product_unable_to_process = nb_product_selected - nb_products_to_process
+        default = super().default_get(fields_list)
+        default.update({
+            'products_to_process': products_to_process,
+            'nb_product_selected': nb_product_selected,
+            'nb_products_to_process': nb_products_to_process,
+            'nb_product_unable_to_process': nb_product_unable_to_process,
+        })
+        return default
+
+    products_to_process = fields.Many2many(
+        comodel_name='product.product',
+        help="The list of selected products that meet the criteria (have a barcode and no image)",
+    )
+    nb_product_selected = fields.Integer(string="Number of selected products", readonly=True)
+    nb_products_to_process = fields.Integer(string="Number of products to process", readonly=True)
+    nb_product_unable_to_process = fields.Integer(
+        string="Number of product unprocessable",
+        readonly=True,
+    )
+
+    def action_fetch_image(self):
+        """ Trigger a cron job to fetch image on Google
+
+        The first ten images are immediately fetched to improve the user experience. This way, they
+        can immediately view the images and be assured that the task is running well. Also, if any
+        error occurs, it can be thrown to the user. Then, a cron job is triggered to be run as soon
+        as possible, unless the daily request limit has been reached. In that case, the cron job
+        will be scheduled to run a day later.
+
+        :return: A notification to inform the user that the task is triggered
+        :rtype: dict
+        :raises UserError: If the project is misconfigured on Google's side
+        :raises UserError: If the API Key or Search Engine ID is incorrect
+        """
+        self.products_to_process.image_fetch_pending = True
+        try:
+            self._fetch_image_from_google(self._get_products_to_process(10))
+        except HTTPError as http_error:
+            if http_error.response.status_code == requests.codes.too_many_requests:
+                self.env.ref('product_images.ir_cron_fetch_image')._trigger(
+                    fields.Datetime.now() + timedelta(days=1.0)
+                )
+                return {
+                    'type': 'ir.actions.client',
+                    'tag': 'display_notification',
+                    'params': {
+                        'type': 'warning',
+                        'title': _("Product images"),
+                        'message': _("Custom search quota exceeded. We will start again tomorrow"),
+                        'next': {'type': 'ir.actions.act_window_close'},
+                    }
+                }
+            elif http_error.response.status_code == requests.codes.forbidden:
+                project_url = re.search(
+                    r"(?P<url>https?://[^\s]+)",
+                    http_error.response.json().get('error').get('message'),
+                ).group("url")
```

Comment:
This is going to blow up if google ever decides to change the information contained in the `message`.

Also `.get('a').get('b')` is doubly broken:

* the first `get` is useless because if the key is not present it'll just break less clearly on the second
* the second `get` is useless because `re.search` does not allow `None` as a "string"

So using `.get` here does not work around any possible error, it just makes them harder to understand.